### PR TITLE
Unit testing frameworks

### PR DESCRIPTION
**Pull request by alex39292**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,35 +1,47 @@
...
+    "version": "1.0.0",
+    "description": "Tasks from EPAM Lab on JavaScript",
+    "main": "index.js",
+    "scripts": {
+      "pretask": "echo \"Pre script\"",
+      "task": "node arrays/task.js",
+      "posttask": "echo \"Post script\"",
+      "test": "mocha",
+      "lint": "eslint ./",
+      "coverage": "nyc npm run test",
```
Comment (SiarheiRylach) [inline package.json]: sugar: nyc npm test

File: unit_testing_framework/calculator/calculator.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    /**
+       *
+       * Sum all parameters of type Number and returns answer
+       * @param {...Number} values
+       * @return {Number} Sum of numbers
+       * @memberof Calculator
+       */
+    add(...values) {
+        if (Array.isArray(values) && (values.length !== 1) &&
```
Comment (SiarheiRylach) [inline unit_testing_framework/calculator/calculator.js]: Why do you use it ```Array.isArray(values)``` ?

File: unit_testing_framework/calculator/calculator.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            values.every(item => typeof item === 'number')) {
+            return values.reduce((a, b) => a * b);
+        } else {
+            throw new TypeError('Invalid types of parameters');
+        }
+    }
+}
+
+module.exports = Calculator
```
Comment (SiarheiRylach) [inline unit_testing_framework/calculator/calculator.js]: ; in the end of line

File: test/add.spec.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+const Calculator = require('../unit_testing_framework/calculator/calculator');
+const {expect} = require('chai');
+
+describe('Tests for method add()', function() {
+    let calculator;
+
+    beforeEach(function() {
```
Comment (SiarheiRylach) [inline test/add.spec.js]: after each with `null` for calculator variable to be sure that every test is independent

File: test/multiply.spec.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    it('Multiply 2 numbers', function() {
+        expect(calculator.multiply(2, 2)).to.be.equal(4);
+    });
+
+    it('Multiply 4 numbers', function() {
+        expect(calculator.multiply(2, 2, 6, 10)).to.be.equal(240);
+    });
+
+    it('Multiply invalid parameters', function() {
+        expect(() => {
```
Comment (SiarheiRylach) [inline test/multiply.spec.js:24]: I'm glad to see that you've found the workaround for catching exceptions 👍

File: package.json (hunk 1 above)
Comment (alex39292) [reply package.json]: Ok)

File: unit_testing_framework/calculator/calculator.js (hunk 2 above)
Comment (alex39292) [reply unit_testing_framework/calculator/calculator.js]: It was to understand if is it) But it's really unnecessary

File: unit_testing_framework/calculator/calculator.js (hunk 3 above)
Comment (alex39292) [reply unit_testing_framework/calculator/calculator.js]: added

File: test/add.spec.js (hunk 4 above)
Comment (alex39292) [reply test/add.spec.js]: Added afterEach() into add.spec.js and multiply.spec.js

**Review by SiarheiRylach — APPROVED** [review]

**Pull request merged**